packages/api/src/ApiRx.ts

```diff
@@ -23,17 +23,31 @@ import {ApiOptions as ApiOptionsBase} from '@plugnet/api/types';
 import {ProviderInterface} from '@plugnet/rpc-provider/types';
 import {isFunction, isObject} from '@plugnet/util';
 
-import {Observable} from 'rxjs';
+import {fromEvent, Observable, race, throwError} from 'rxjs';
+import {switchMap, timeout} from 'rxjs/operators';
+import {DEFAULT_TIMEOUT} from './Api';
 import * as derives from './derives';
 import staticMetadata from './staticMetadata';
 import {ApiOptions, IPlugin} from './types';
 import {getProvider} from './util/getProvider';
 import logger from './util/logging';
+
 const Types = require('@cennznet/types');
 
 export class ApiRx extends ApiRxBase {
     static create(options: ApiOptions | ProviderInterface = {}): Observable<ApiRx> {
-        return new ApiRx(options).isReady;
+        const apiRx = new ApiRx(options);
+        const timeoutMs =
+            isObject(options) && isFunction((options as ProviderInterface).send)
+                ? undefined
+                : (options as ApiOptions).timeout;
+
+        return race(
+            apiRx.isReady.pipe(timeout(timeoutMs ? timeoutMs : DEFAULT_TIMEOUT)),
+            fromEvent((apiRx as any)._eventemitter, 'error').pipe(
+                switchMap(err => throwError(new Error('Connection fail')))
```

Comment:
changed to use throwError to throw an Error object here, error message `Connection fail`